modules/mcuboot/CMakeLists.txt

```diff
@@ -707,6 +707,59 @@ CONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT cannot be used with CONFIG_XIP_SPLIT_IMAGE.
       endif()
     endif()
 
+    # Targets for adding nrf70 wifi firmware patch. This workflow creates the nrf70_signed.hex and nrf70_update.bin files that can be
+    # used in DFU process, and multi-image DFU process.
+    if(CONFIG_NRF_WIFI_FW_PATCH_DFU)
+
+      set(nrf70_binary_name nrf70_update.bin)
+
+      sign(
+        SIGNED_BIN_FILE_IN ${PROJECT_BINARY_DIR}/nrf70.hex
+        SIGNED_HEX_FILE_NAME_PREFIX ${PROJECT_BINARY_DIR}/nrf70
+        SLOT_SIZE $<TARGET_PROPERTY:partition_manager,PM_MCUBOOT_PRIMARY_SIZE>
+        SIGNED_HEX_FILE_OUT nrf70_signed_hex
+        SIGNED_BIN_FILE_OUT nrf70_signed_bin
+        SIGNED_HEX_TEST_FILE_OUT nrf70_signed_test_hex
+        DEPENDS ${PROJECT_BINARY_DIR}/nrf70.hex
+      )
+
+      add_custom_target(mcuboot_nrf70_target
+        DEPENDS
+        ${nrf70_signed_hex}
+        ${nrf70_signed_bin}
+        ${nrf70_signed_test_hex}
+      )
+
+      set_property(GLOBAL PROPERTY
+        nrf70_wifi_fw_PM_HEX_FILE
+        ${nrf70_signed_hex}
+      )
+
+      set_property(GLOBAL PROPERTY
+        nrf70_wifi_fw_PM_TARGET
+        mcuboot_nrf70_target
+      )
+
+      add_dependencies(
+        mcuboot_sign_target
+        mcuboot_nrf70_target
+      )
+
+      # Add nrf70 update file to the existing bin files list before generating the zip file.
+      list(APPEND generate_bin_files
+        ${PROJECT_BINARY_DIR}/${nrf70_binary_name}
+      )
+
+      # Add nrf70-related sctipt params before generating the zip file.
+      list(APPEND generate_script_params
+        "${nrf70_binary_name}load_address=$<TARGET_PROPERTY:partition_manager,PM_NRF70_WIFI_FW_ADDRESS>"
+        "${nrf70_binary_name}image_index=2"
```

Comment:
note that this conflicts with qspi xip @nvlsianpu so you'll never be able to use both of these going forward in ncs

Reply:
That's right, @nvlsianpu we can discuss how to make the possibility to use both solutions.

Reply:
@nordicjm my thinking is that our clients should still be able to use XIP configuration with nRF7002 patch placed on external flash but indeed in slightly different configuration. The third image would be used for executable FW on external flash AND the nRF7002 patch that be defined as code - which would effectively mean [this](https://github.com/nrfconnect/sdk-nrf/blob/main/drivers/wifi/nrf700x/src/fw_load.c#L270C22-L270C38) code is executed and not [this](https://github.com/nrfconnect/sdk-nrf/blob/main/drivers/wifi/nrf700x/src/fw_load.c#L145). Does it make sense, or there are some other blockers - at least conceptually (maybe some changes are needed in this PR).

Reply:
@ArekBalysNordic Discussed with @LuDuda off-channel: Today let's concentrate on what you provide - mutual exclusion of this feature witch qspi-ext-xip is acceptable (worth to introduce Cmake assertion on mutual selection of theses fatures). Steps needed for introduction coexistence of these features should be considered and planed after the 2.6.0 release.

Reply:
Eventually we can use "image_index=3" (align slot indexes as well)  in case CONFIG_XIP_SPLIT_IMAG is selected. I think along with proper pm_static that all which are needed.